src/js/histogram.js

```diff
@@ -269,140 +287,215 @@ function drawHistogram(data, containerId, options) {
-    title: {
-      text: `${options.lens ? `${options.lens.name}: ` : '' }` + `Histogram of ${options.name}`,
-      style: {
-        "font-weight": "normal"
-      }
-    },
-    subtitle: {
-      text: `Source: <a href="http://httparchive.org">httparchive.org</a> (${prettyDate(options.date)})`,
-      useHTML: true
-    },
-    plotOptions: {
-      column: {
-        grouping: false
-      },
-      series: {
-        events: {
-          // Keep CDF visibility in sync.
-          hide: function() {
-            const cdf = chart.series.find(s => s.name === `${this.name} CDF`);
-            if (cdf) {
-              cdf.hide();
+  console.log('Options:', options);
+  console.log('Series:', series);
+
+  const axis = options.xaxis;
+  const chartData = [];
+  chartData.labels = axis.data;
+  chartData.datasets = series;
+
+  const chart = new Chart(
+    document.getElementById(options.chartId),
+    {
+      options: {
+        responsive: true,
+        maintainAspectRatio: false,
+        layout: {
+          padding: {
+            bottom: 20,
+          },
+        },
+        legend: {
+          labels: {
+            padding: {
+                top: 20,
+                bottom: 20,
+            },
+          },
+        },
+        plugins: {
+          title: {
+            display: true,
+            text: options.lens ? `${options.lens.name}: ` : '' + `Histogram of ${options.name}`,
+            font: {
+              family: 'Helvetica, Arial, sans-serif',
+              size: 16,
+              weight: 'normal',
+            },
+            padding: {
+                top: 20,
+            },
+          },
+          subtitle: {
+            display: true,
+            text: `Source: httparchive.org (${prettyDate(options.date)})`,
+            font: {
+              family: 'Helvetica, Arial, sans-serif',
+              size: 13,
+              weight: 'normal',
+            },
+            padding: {
+                bottom: 80,
+            },
+          },
+          tooltip: {
+            mode: 'nearest',
+            axis: 'x',
+            intersect: false,
+            titleAlign: 'center',
+            callbacks: {
+              title: function (context) {
+                return `${context[0]?.label} ${options.type}`;
+              },
+              label: function(context) {
+                if (context.dataset?.tooltip?.enabled == false) {
+                  return null;
+                }
+                if (context.dataset.type === 'bar') {
+                  return ` ${context.dataset.label}: ${context.raw.y.toFixed(2)}%`;
+                }
+                return ` ${context.dataset.label}: ${context.raw[1].toFixed(0)}%`;
+              },
             }
           },
-          show: function() {
-            const cdf = chart.series.find(s => s.name === `${this.name} CDF`);
-            if (cdf) {
-              cdf.show();
+          legend: {
+            display: true,
+            position: 'bottom',
```

Comment:
Not related to this PR, but a general observation:
The legend on the top immediately informs viewers about the categories before they look at the chart. VS when it's in the bottom, it interrupts the flow (to scroll and find), makes confused.

```suggestion
            position: 'top',
```
Also timeseries.

And I know we repeat them for every chart, but it's impossible to remember the colours.

Reply:
That makes it a little trickier with the extra buttons that overlay the chart. Not impossible but a little trickier. Will leave this comment open but might merge without that. Will see...